### PR TITLE
uni-31-update: update email authentication on homepage

### PR DESCRIPTION
**Pull request by vivianchengg**

### CODE REVIEW

File: frontend/src/pages/HomePage.tsx [hunk 1]
```diff
@@ -22,28 +23,39 @@ export const HomePage = () => {
   const [loading, setLoading] = useState(false);
   const [open, setOpen] = useState(false);
   const [loginErrorMsg, setLoginErrorMsg] = useState('');
+  const name = accounts[0] && accounts[0].name;
```
Comment (Siyer2) [inline frontend/src/pages/HomePage.tsx]: It seems that this constant isn't used. Can we remove it?

File: frontend/src/pages/HomePage.tsx [hunk 2]
```diff
@@ -22,28 +23,39 @@ export const HomePage = () => {
...
 
   useEffect(() => {
     async function signInSignUp() {
-      const response = await requestMSAuthResult(instance, accounts[0]);
-      console.log('logging in', response.idToken);
+      const isValidEmail =
+        /^[A-Za-z0-9._%+-]+@(unsw.edu.au|unswalumni.com|ad.unsw.edu.au|zmail.unsw.edu.au|student.unsw.edu.au)$/.test(
+          accounts[0].username
+        );
```
Comment (Siyer2) [inline frontend/src/pages/HomePage.tsx]: Files that end in .tsx should predominantly be used for frontend structural stuff - like buttons and textfields. Let's move this into a .ts file. In the `functions` folder, create a new file called `emailValidation.ts`. There, you should create a function that takes in an email string and returns a true or false boolean:
```
export function isValidEmail(email: string): boolean {
  // add your code here
}
```

**Review by Siyer2 — CHANGES_REQUESTED** [review]

File: frontend/src/pages/HomePage.tsx (hunk 1 above)
Comment (vivianchengg) [reply frontend/src/pages/HomePage.tsx]: got it

File: frontend/src/pages/HomePage.tsx (hunk 2 above)
Comment (vivianchengg) [reply frontend/src/pages/HomePage.tsx]: done!

**Review by Siyer2 — APPROVED** [review]
Awesome work!

**Pull request merged**